FEM/rf_st_new.cpp

```diff
@@ -1805,40 +1805,12 @@ void CSourceTerm::FaceIntegration(CRFProcess* pcs, std::vector<long> const& node
 		G2L[k] = i;
 	}
 
-	//----------------------------------------------------------------------
-	// NW 15.01.2010
-	// 1) search element faces on the surface
-	// 2) face integration
-
-	// init
-	for (i = 0; i < (long)msh->ele_vector.size(); i++)
-	{
-		msh->ele_vector[i]->selected = 0; // TODO can use a new variable
-	}
 	std::set<long> set_nodes_on_sfc; // unique set of node id on the surface
 	for (i = 0; i < (long)nodes_on_sfc.size(); i++)
 	{
 		set_nodes_on_sfc.insert(nodes_on_sfc[i]);
 	}
 
-	// filtering elements: elements should have nodes on the surface
-	// Notice: node-elements relation has to be constructed beforehand
-	// CB THMBM
-	// this->getProcess()->CheckMarkedElement(); // CB added to remove bug with deactivated Subdomains
-	// Already added to Set(..)
-	std::vector<long> vec_possible_elements;
```

Comment:
is it really necessary to remove this could? Without this, the computation can be quite slow for large problems.

Reply:
Recovered but enabled all elements to be searched because that surface can be defined inside a domain. The following condition is for the case of   surface being defined inside domain.
```
			// Not a surface face
			if (elem->GetDimension() == e_nei->GetDimension())
				fac = 0.5;
```